skylog/storage/cloudspanner/tree.go

```diff
@@ -0,0 +1,99 @@
+// Package cloudspanner provides implementation of the Skylog storage API in
+// Cloud Spanner.
+package cloudspanner
+
+import (
+	"context"
+
+	"cloud.google.com/go/spanner"
+	"github.com/google/trillian/merkle/compact"
+	"github.com/google/trillian/skylog/storage"
+)
+
+// TreeStorage allows reading and writing from a tree storage.
+type TreeStorage struct {
+	c    *spanner.Client
+	id   int64
+	opts TreeOpts
+}
+
+// NewTreeStorage returns a new TreeStorage for the specified tree and options.
+func NewTreeStorage(c *spanner.Client, treeID int64, opts TreeOpts) *TreeStorage {
+	return &TreeStorage{c: c, id: treeID, opts: opts}
```

Comment:
It looks to me that the opts used for a tree must not change over time? If that's correct then they should maybe be stored in the database.